cli/npm/managed/registry.rs

```diff
@@ -135,8 +135,8 @@ impl CliNpmRegistryApiInner {
     name: &str,
   ) -> Result<Option<Arc<NpmPackageInfo>>, AnyError> {
     let (created, future) = {
-      let mut mem_cache = self.mem_cache.lock();
-      match mem_cache.get(name) {
+      let item = { self.mem_cache.lock().get(name).cloned() };
```

Comment:
These changes are to avoid holding the lock across the new await point. I was between releasing the lock and switching to a `tokio::sync::Mutex`, but haven't benchmarked the tokio mutex yet